### DISCUSSION TITLE
Warning: Solver not applicable (gurobi)

### DISCUSSION THREAD

**Discussion by ellensonny** [yalmip/YALMIP#1406] [Optimization modelling] [answered]
hello Prof,
Power balance constraints:
![qiz1](https://github.com/yalmip/YALMIP/assets/130203697/2f8405ef-975c-49cf-a05a-f4436bf8d366)
and there is part of my codes:
![qiz3](https://github.com/yalmip/YALMIP/assets/130203697/1398e95a-fbd1-4778-b0f2-441aa8f165a1)
![qiz2](https://github.com/yalmip/YALMIP/assets/130203697/815c46e9-e1cb-40e8-8c3c-ed0ce636051c)

but the result says like "Warning: Solver not applicable (gurobi)".Doesn't my code correctly express the constraints of the mathematical model? If so,how to solve it? Thanks for your reply.

**Comment by johanlofberg** [comment]
you are using trigonometric functions which gurobi doesn't support

**Reply by ellensonny** [reply]
So how to solve this problem generally ,Prof? Modify the mathematical model to eliminate trigonometric functions, or change to a suitable solver?

**Reply by johanlofberg** [reply, marked as answer]
A better model is always best of course.

I don't think there is any other global solver in YALMIP except the built-in bmibnb to handle everything in the model. Hence, very very hard and likely not possible to solve

**Reply by ellensonny** [reply]
ok.Thank you very much for reply.
